x-pack/legacy/plugins/index_management/public/components/json_editor/use_json.ts

```diff
@@ -10,7 +10,10 @@ import { i18n } from '@kbn/i18n';
 import { isJSON } from '../../../../../../../src/plugins/es_ui_shared/static/validators/string';
 
 export type OnUpdateHandler<T = { [key: string]: any }> = (arg: {
-  getData(): T;
+  data: {
```

Comment:
@jloleysens I slightly changed the contract to get the json data in order to align with the contract of the form hook update.